spec/models/user_spec.rb

```diff
@@ -1,5 +1,24 @@
 require 'rails_helper'
 
-RSpec.describe User, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+describe User do
+  it { should have_valid(:username).when('username') }
+  it { should have_valid(:username).when('anothername123') }
+  it { should_not have_valid(:username).when('') }
+  it { should_not have_valid(:username).when(nil) }
+  it { should_not have_valid(:username).when('with space') }
+  it { should_not have_valid(:username).when('upsernamethatgoesonforever') }
+  it { should_not have_valid(:username).when('with#$%^characters') }
+
+  it { should have_valid(:email).when('user@example.com', 'another@gmail.com') }
+  it { should_not have_valid(:email).when(nil, '', 'user', 'users@com',
```

Comment:
Avoid using {...} for multi-line blocks.<br>Block body expression is on the same line as the block start.